pandas/tests/frame/methods/test_first_and_last.py

```diff
@@ -95,3 +99,48 @@ def test_empty_not_input(self, func):
         result = getattr(df, func)(offset=1)
         tm.assert_frame_equal(df, result)
         assert df is not result
+
+    @pytest.mark.parametrize("start, periods", [("2010-03-31", 1), ("2010-03-30", 2)])
+    def test_first_with_first_day_last_of_m_DO(self, frame_or_series, start, periods):
+        x = frame_or_series([1] * 100, index=bdate_range(start, periods=100))
```

Comment:
just out of interest, any reason to use `bdate_range` instead of `date_range`? the latter might be easier to read (I had to check whether `2010-03-31` as a business date)

for the title, perhaps
```
test_last_day_of_month_with_date_offset
```
? the file is already `test_first_and_last` so maybe we don't need `first` in the title again (even though some other tests have it 😄 )

Reply:
I used bdate_range to do it like the existing/previous tests. Same with the name. 

Changing it now..